client/src/controllers/DialogController.test.js

```diff
@@ -0,0 +1,141 @@
+            document.addEventListener('w-dialog:hidden', hiddenListener);
+
+            application.start();
+
+            await Promise.resolve();
+
+            expect(shownListener).not.toHaveBeenCalled();
+            expect(hiddenListener).not.toHaveBeenCalled();
+
+            const dialog = document.getElementById('dialog-container');
+
+            // closed by default
+            expect(dialog.getAttribute('aria-hidden')).toEqual('true');
+            expect(document.documentElement.style.overflowY).toBe('');
+
+            // show the dialog manually
+            dialog.dispatchEvent(new CustomEvent('w-dialog:show'));
+
+            expect(dialog.getAttribute('aria-hidden')).toEqual(null);
+            expect(shownListener).toHaveBeenCalledWith(
+                expect.objectContaining({
+                    detail: expect.objectContaining({
+                        body: document.getElementById('dialog-body'),
+                        dialog: expect.any(Object),
+                    }),
+                }),
+            );
+            expect(hiddenListener).not.toHaveBeenCalled();
+            // add style to root element on shown by default
+            expect(document.documentElement.style.overflowY).toBe('hidden');
+
+            // hide the dialog manually
+            dialog.dispatchEvent(new CustomEvent('w-dialog:hide'));
+
+            expect(dialog.getAttribute('aria-hidden')).toEqual('true');
+            expect(shownListener).toHaveBeenCalledTimes(1);
+            expect(hiddenListener).toHaveBeenCalledWith(
+                expect.objectContaining({
+                detail: expect.objectContaining({
+                    body: document.getElementById('dialog-body'),
+                    dialog: expect.any(Object),
+                }),
+                }),
+            );
+            // reset style on root element when hidden by default
+            expect(document.documentElement.style.overflowY).toBe('');
+        });
+
+        it('should support the ability use a theme to avoid document style change', async () => {
+        // update the DOM with the theme value
+            application.start();
+
+            await Promise.resolve();
+
+            const dialog = document.getElementById('dialog-container');
+
+            // closed by default
+            expect(dialog.getAttribute('aria-hidden')).toEqual('true');
+
+            dialog.dispatchEvent(new CustomEvent('w-dialog:show'));
+
+            expect(dialog.getAttribute('aria-hidden')).toEqual(null);
+
+            dialog.dispatchEvent(new CustomEvent('w-dialog:hide'));
+
+            expect(dialog.getAttribute('aria-hidden')).toEqual('true');
+
+            dialog.dispatchEvent(new CustomEvent('w-dialog:show'));
+
+            // adding a theme value to remove the default behaviour
+            dialog.setAttribute("class", "w-dialog--floating");
+            dialog.setAttribute("data-w-dialog-theme-value", "floating");
+
```

Comment:
When making a data value change, you may need to ensure that all DOM behaviour has resolved.

Try ...

```
await Promise.resolve();
```

Reply:
Also, when you set this value, the dialog code has already run.

You will need to set the value before you do application.start.